node/lib/util/tree_util.js

```diff
@@ -191,3 +193,83 @@ exports.writeTree = co.wrap(function *(repo, baseTree, changes) {
     });
     return yield writeSubtree(baseTree, directory);
 });
+
+/**
+ * Return an blob ID for the specified `filename` in the specified `repo`.
+ *
+ * @param {NodeGit.Repository} repo
+ * @param {String}             filename
+ * @return {NodeGit.Oid}
+ */
+exports.hashFile = function (repo, filename) {
+    assert.instanceOf(repo, NodeGit.Repository);
+    assert.isString(filename);
+
+    // 'createFromDisk' is unfinished; instead of returning an id, it
+    // takes an ID object and writes into it, unlike the rest of its
+    // brethern on `Blob`.  TODO: patch nodegit with corrected API.
+
+    const placeholder =
+            NodeGit.Oid.fromString("0000000000000000000000000000000000000000");
+    NodeGit.Blob.createFromDisk(placeholder, repo, filename);
+    return placeholder;
+};
+
+/**
+ * Return a map from path to `Change` for the working directory of the
+ * specified `repo` having the specified `status`.  If the specified `all` is
+ * true, include unstaged changes.
+ *
+ * @param {NodeGit.Repository} repo
+ * @param {RepoStatus}         status
+ * @param {Boolean}            all
```

Comment:
yeah, you're right. I saw an `all` option in `git stash`, but it means to also include ignored files